### PR TITLE
Add integration test on query-scheduler

### PR DESCRIPTION
**Pull request by pracucci**
**What this PR does**:
In this PR I've added integration tests for the query-scheduler. The diff will be easier to review if you enable "hide whitespace changes".

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/querier/frontend2/querier_scheduler_worker.go [hunk 1]
```diff
@@ -83,8 +83,8 @@ func NewQuerierSchedulerWorkers(cfg QuerierWorkersConfig, handler RequestHandler
 	}
 
 	frontendClientsGauge := promauto.With(reg).NewGauge(prometheus.GaugeOpts{
-		Name: "cortex_query_scheduler_worker_frontend_clients",
-		Help: "The current number of frontend clients.",
+		Name: "cortex_querier_query_frontend_clients",
```
Comment (pracucci) [inline pkg/querier/frontend2/querier_scheduler_worker.go:86]: I would prefer it `cortex_query_scheduler` metrics prefix are reserved for the scheduler. Thoughts?

**Comment by pracucci** [comment]
Flaky test should be fixed by this PR: https://github.com/cortexproject/cortex/pull/3434

File: pkg/querier/frontend2/querier_scheduler_worker.go (hunk 1 above)
Comment (pstibrany) [reply pkg/querier/frontend2/querier_scheduler_worker.go:86]: :+1: Good idea, thanks for spotting.

**Review by pstibrany — APPROVED** [review]
Great job, thank you very much!

**Pull request merged**